WcaOnRails/app/models/locale.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+class Locale < SimpleDelegator
+  include ActionView::Helpers::TextHelper
+  SOME_CHARS = '[\s\S]*?'
```

Comment:
Some numbers, constructing one translation locale (out of the rails structure), with the `Regexp::MULTILINE` enabled:
(I truncated the output when the calls added nothing to the cumulative time, therefore the last cumulative time is the total time)
<details>
  <summary>Using `.*`:</summary>

>  %   cumulative   self              self     total
 time   seconds   seconds    calls  ms/call  ms/call  name
 94.31     6.13      6.13      695     8.82     8.82  String#sub
  0.77     6.18      0.05      112     0.45     3.21  Kernel#require
  0.62     6.22      0.04     1392     0.03     0.85  Psych::Visitors::ToRuby#accept
  0.62     6.26      0.04     1263     0.03     0.07  Psych::Visitors::ToRuby#deserialize
  0.46     6.29      0.03     1267     0.02    38.43  Locale#decorate_with_hashes
  0.46     6.32      0.03     1392     0.02     0.81  Psych::Visitors::Visitor#visit
  0.46     6.35      0.03     1263     0.02     0.02  Psych::TreeBuilder#scalar
  0.31     6.37      0.02      657     0.03     0.05  Psych::ScalarScanner#tokenize
  0.31     6.39      0.02      695     0.03     0.03  Regexp#initialize
  0.15     6.40      0.01     1264     0.01     0.01  BasicObject#!
  0.15     6.41      0.01      130     0.08     0.08  Array#push
  0.15     6.42      0.01        1    10.00    60.00  Psych::Parser#parse
  0.15     6.43      0.01      667     0.01     0.01  Regexp#===
  0.15     6.44      0.01     1263     0.01     0.08  Psych::Visitors::ToRuby#visit_Psych_Nodes_Scalar
  0.15     6.45      0.01     1392     0.01     0.82  Psych::Visitors::Visitor#accept
  0.15     6.46      0.01      128     0.08     6.33  Psych::Visitors::ToRuby#visit_Psych_Nodes_Mapping
  0.15     6.47      0.01      128     0.08     0.08  Psych::TreeBuilder#end_mapping
  0.15     6.48      0.01     1263     0.01     0.02  Psych::ClassLoader#load
  0.15     6.49      0.01      695     0.01     0.01  Locale#leaf?
  0.15     6.50      0.01      128     0.08   190.23  Hash#each
</details>

<details>
  <summary>Using `.*?`:</summary>

>  %   cumulative   self              self     total
 time   seconds   seconds    calls  ms/call  ms/call  name
 22.81     0.13      0.13      695     0.19     0.20  String#sub
 12.28     0.20      0.07     1518     0.05     1.46  Locale#decorate_with_hashes
 10.53     0.26      0.06      823     0.07     1.88  Psych::Visitors::ToRuby#revive_hash
  7.02     0.30      0.04     1263     0.03     0.05  Psych::TreeBuilder#scalar
  7.02     0.34      0.04      112     0.36     2.68  Kernel#require
  5.26     0.37      0.03     1263     0.02     0.04  Psych::Visitors::ToRuby#visit_Psych_Nodes_Scalar
  5.26     0.40      0.03      695     0.04     0.04  Locale#leaf?
  3.51     0.42      0.02     2101     0.01     0.27  Class#new
  3.51     0.44      0.02     1706     0.01     0.01  Hash#[]=
  3.51     0.46      0.02     1392     0.01     0.74  Psych::Visitors::Visitor#visit
  1.75     0.47      0.01      196     0.05     3.98  Array#each
  1.75     0.48      0.01     1263     0.01     0.01  Psych::Nodes::Scalar#initialize
  1.75     0.49      0.01     1263     0.01     0.01  Psych::Visitors::ToRuby#resolve_class
  1.75     0.50      0.01     1263     0.01     0.02  Psych::Visitors::ToRuby#deserialize
  1.75     0.51      0.01     1392     0.01     0.75  Psych::Visitors::Visitor#accept
  1.75     0.52      0.01     1392     0.01     0.01  Hash#empty?
  1.75     0.53      0.01      695     0.01     0.01  Regexp#initialize
  1.75     0.54      0.01      128     0.08     0.08  Psych::TreeBuilder#start_mapping
  1.75     0.55      0.01     1391     0.01     0.01  Array#<<
  1.75     0.56      0.01        1    10.00    80.00  Psych::Parser#parse
  1.75     0.57      0.01     1392     0.01     0.76  Psych::Visitors::ToRuby#accept
</details>

<details>
<summary>Using `[.]*`:</summary>

>  %   cumulative   self              self     total
 time   seconds   seconds    calls  ms/call  ms/call  name
 62.90     0.78      0.78      695     1.12     1.14  String#sub
  4.84     0.84      0.06      825     0.07     8.93  Locale#decorate_with_hashes
  3.23     0.88      0.04     2101     0.02     0.59  Class#new
  2.42     0.91      0.03     1263     0.02     0.03  Psych::TreeBuilder#scalar
  2.42     0.94      0.03     1392     0.02     0.78  Psych::Visitors::Visitor#visit
  1.61     0.96      0.02      657     0.03     0.05  Psych::ScalarScanner#tokenize
  1.61     0.98      0.02      695     0.03     0.04  Locale#leaf?
  1.61     1.00      0.02      112     0.18     3.30  Kernel#require
  1.61     1.02      0.02      223     0.09     0.09  Gem.find_unresolved_default_spec
  1.61     1.04      0.02        1    20.00    70.00  Psych::Parser#parse
  1.61     1.06      0.02     2783     0.01     0.01  Hash#[]
  1.61     1.08      0.02     1263     0.02     0.06  Psych::Visitors::ToRuby#visit_Psych_Nodes_Scalar
  0.81     1.09      0.01       13     0.77     0.77  Array#map
  0.81     1.10      0.01     1264     0.01     0.01  BasicObject#!
  0.81     1.11      0.01     2087     0.00     0.00  Delegator.const_missing
  0.81     1.12      0.01      128     0.08     0.08  Psych::TreeBuilder#start_mapping
  0.81     1.13      0.01     1395     0.01     0.01  Kernel#hash
  0.81     1.14      0.01     1706     0.01     0.01  Hash#[]=
  0.81     1.15      0.01     1263     0.01     0.02  Psych::Visitors::ToRuby#resolve_class
  0.81     1.16      0.01      667     0.01     0.01  Regexp#===
  0.81     1.17      0.01     1392     0.01     0.80  Psych::Visitors::ToRuby#accept
  0.81     1.18      0.01      823     0.01     1.83  Psych::Visitors::ToRuby#revive_hash
  0.81     1.19      0.01      128     0.08     5.94  Enumerable#each_slice
  0.81     1.20      0.01      128     0.08     6.02  Psych::Visitors::ToRuby#visit_Psych_Nodes_Mapping
  0.81     1.21      0.01      695     0.01     0.01  Regexp#initialize
  0.81     1.22      0.01       34     0.29     0.29  nil#
  0.81     1.23      0.01     1392     0.01     0.79  Psych::Visitors::Visitor#accept
</details>

<details>
<summary>Using `[.]*?`:</summary>

>  %   cumulative   self              self     total
 time   seconds   seconds    calls  ms/call  ms/call  name
 70.63     0.89      0.89      695     1.28     1.28  String#sub
  3.97     0.94      0.05      825     0.06     9.53  Locale#decorate_with_hashes
  3.17     0.98      0.04      112     0.36     3.66  Kernel#require
  2.38     1.01      0.03      657     0.05     0.06  Psych::ScalarScanner#tokenize
  2.38     1.04      0.03     1263     0.02     0.02  Psych::TreeBuilder#scalar
  1.59     1.06      0.02      128     0.16     0.16  Psych::TreeBuilder#start_mapping
  1.59     1.08      0.02     1263     0.02     0.08  Psych::Visitors::ToRuby#deserialize
  1.59     1.10      0.02      823     0.02     1.80  Psych::Visitors::ToRuby#revive_hash
  1.59     1.12      0.02      196     0.10     3.88  Array#each
  1.59     1.14      0.02     1392     0.01     0.77  Psych::Visitors::Visitor#visit
  0.79     1.15      0.01      133     0.08     0.08  Array#last
  0.79     1.16      0.01        7     1.43     1.43  Gem::Specification.gemspec_stubs_in
  0.79     1.17      0.01     1099     0.01     0.01  Hash#key?
  0.79     1.18      0.01     1264     0.01     0.01  BasicObject#!
  0.79     1.19      0.01     1393     0.01     0.01  Kernel#class
  0.79     1.20      0.01        1    10.00    70.00  Psych::Parser#parse
  0.79     1.21      0.01     1263     0.01     0.02  Psych::ClassLoader#load
  0.79     1.22      0.01     1263     0.01     0.02  Psych::Visitors::ToRuby#resolve_class
  0.79     1.23      0.01     1263     0.01     0.09  Psych::Visitors::ToRuby#visit_Psych_Nodes_Scalar
  0.79     1.24      0.01      695     0.01     0.01  Locale#leaf?
  0.79     1.25      0.01      695     0.01     0.01  Regexp#initialize
  0.79     1.26      0.01     2783     0.00     0.00  Hash#[]
</details>

<details>
<summary>Our current solution (`[\s\S]*?` without multiline mode):</summary>

>  %   cumulative   self              self     total
 time   seconds   seconds    calls  ms/call  ms/call  name
 34.48     0.20      0.20      695     0.29     0.30  String#sub
 10.34     0.26      0.06     1518     0.04     1.55  Locale#decorate_with_hashes
  5.17     0.29      0.03     1263     0.02     0.03  Psych::TreeBuilder#scalar
  5.17     0.32      0.03     2783     0.01     0.01  Hash#[]
  5.17     0.35      0.03      657     0.05     0.06  Psych::ScalarScanner#tokenize
  5.17     0.38      0.03     1263     0.02     0.02  Psych::Visitors::ToRuby#resolve_class
  3.45     0.40      0.02     1392     0.01     0.75  Psych::Visitors::Visitor#visit
  3.45     0.42      0.02     1392     0.01     0.77  Psych::Visitors::ToRuby#accept
  3.45     0.44      0.02      823     0.02     1.79  Psych::Visitors::ToRuby#revive_hash
  1.72     0.45      0.01      196     0.05     3.78  Array#each
  1.72     0.46      0.01       19     0.53     0.53  Array#map
  1.72     0.47      0.01      112     0.09     3.39  Kernel#require
  1.72     0.48      0.01     1353     0.01     0.01  String#empty?
  1.72     0.49      0.01      128     0.08     0.16  Psych::TreeBuilder#start_mapping
  1.72     0.50      0.01     1263     0.01     0.07  Psych::Visitors::ToRuby#visit_Psych_Nodes_Scalar
  1.72     0.51      0.01      695     0.01     0.01  Locale#leaf?
  1.72     0.52      0.01       56     0.18     0.18  MonitorMixin#mon_enter
  1.72     0.53      0.01       30     0.33     0.33  Module#attr_accessor
  1.72     0.54      0.01     2101     0.00     0.27  Class#new
  1.72     0.55      0.01        6     1.67     1.67  Gem::Specification.dirs
  1.72     0.56      0.01      128     0.08     0.08  Psych::TreeBuilder#end_mapping
  1.72     0.57      0.01      130     0.08     0.08  Psych::Nodes::Node#initialize
  1.72     0.58      0.01      695     0.01     0.01  Regexp#initialize

</details>

I didn't realize making the match optional would make it so much faster!
I'll change the expression to `.*?`, it's the same execution time but should be less confusing.